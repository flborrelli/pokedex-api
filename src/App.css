* {
  margin: 0;
  padding: 0;
}

html {
  font-size: 10px;
}

/* NAVBAR */

.mynavbar {
  background-color: tomato;
  box-shadow: 0px 3px 6px 0px rgba(0,0,0,0.75);
}

.navbar-title {
  letter-spacing: 0.1rem;
  text-align: center;
  font-size: 1.5rem;
  color: rgba(255, 255, 255, 0.9)
}

.nav-img {
  width: 10vw;
}

/* SEARCHBAR */

.search-bar {
  text-align: center;
}

.icon {
  width: 50vw;
}

/* POKEMONLIST */


.segment {
  width: 50vw;
}

.sad-pokemon{
  width: 10vw;
}

/* LOADER */
.loader {
  width: 7vw;
  animation: frames 2s  linear 0s infinite;
}

.loading-text {
  text-align: center;
  font-size: 1.6rem;
}

/* AnimationFrames */
@keyframes frames{
  0% {
    transform: rotate(0deg);
  }
  100%{
    transform: rotate(360deg);
  }
}