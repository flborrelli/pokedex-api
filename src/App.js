import React, { useState, useEffect, useMemo} from 'react';
import './App.css';
import NavBar from './components/NavBar';
import PokemonList from './components/PokemonList';
import axios from 'axios';

function App() {
  const [pokemons, setPokemons] = useState([]);
  const [pokemonInfos, setPokemonInfos] = useState({
    id: '',
    avatar: ''
  })


  useEffect(() => {
    getAPIData()
  }, []);


  const getAPIData = () => {
    const rootAPI_URL = 'https://pokeapi.co/api/v2/pokemon?limit=20';
    axios.get(rootAPI_URL)
    .then(responseFromAPI => {
      const pokemonArray = responseFromAPI.data.results;
      console.log(pokemonArray)
      pokemonArray.map(pokemon => {
        getPokemonInfos(pokemon)
      })
      setPokemons(pokemonArray.map(e => e.name))
    })
    .catch(error => console.log('An error occured while fetching API:', error))
  }

  const getPokemonInfos = (pokemon) => {
    const url = pokemon.url;
    axios.get(url)
    .then(pokemonInfos => {
      console.log(pokemonInfos)
    })
    .catch(error => error => console.log('An error occured while accessing Pokemon Infos:', error))
  }

  return (
    <div>
      <NavBar />
      <PokemonList getPokemons={pokemons}/>
      {/* {pokemons.map(e => e)} */}
    </div>
  );
}

export default App;
