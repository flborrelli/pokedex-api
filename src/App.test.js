import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { create } from 'react-test-renderer';

test('it should render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('should match with app snapshot', () => {
  const appComponent = create(<App />);
  expect(appComponent.toJSON()).toMatchSnapshot();
})
