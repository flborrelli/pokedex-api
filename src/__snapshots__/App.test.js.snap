// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match with snapshot 1`] = `
<div
  className="home"
>
  <nav
    className="mynavbar d-flex justify-content-center navbar navbar-expand navbar-light"
    data-testid="navbar"
  >
    <a
      className="logo mr-4 navbar-brand"
      href="#"
    >
      <img
        alt=""
        className="nav-img d-inline-block align-top"
        src="/images/pokeapi.png"
      />
    </a>
    <h2
      className="navbar-title ml-3"
    >
      Find your favourites Pokemons
    </h2>
  </nav>
  <div
    className="search-bar mt-5"
  >
    <div
      className="ui icon input"
    >
      <input
        onChange={[Function]}
        placeholder="Search a Pokemon..."
        type="text"
      />
      <i
        aria-hidden="true"
        className="search icon"
      />
    </div>
  </div>
  <div
    className="d-flex justify-content-center mt-5 mb-4"
  >
    <div
      className="pokelist"
    >
      <div>
        <img
          alt="loader"
          className="loader"
          src="/images/pokeball.png"
        />
      </div>
      <div
        className="mt-3 loading-text"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;

exports[`should match with app snapshot 1`] = `
<div
  className="home"
>
  <nav
    className="mynavbar d-flex justify-content-center navbar navbar-expand navbar-light"
    data-testid="navbar"
  >
    <a
      className="logo mr-4 navbar-brand"
      href="#"
    >
      <img
        alt=""
        className="nav-img d-inline-block align-top"
        src="/images/pokeapi.png"
      />
    </a>
    <h2
      className="navbar-title ml-3"
    >
      Find your favourites Pokemons
    </h2>
  </nav>
  <div
    className="search-bar mt-5"
  >
    <div
      className="ui icon input"
    >
      <input
        onChange={[Function]}
        placeholder="Search a Pokemon..."
        type="text"
      />
      <i
        aria-hidden="true"
        className="search icon"
      />
    </div>
  </div>
  <div
    className="d-flex justify-content-center mt-5 mb-4"
  >
    <div
      className="pokelist"
    >
      <div>
        <img
          alt="loader"
          className="loader"
          src="/images/pokeball.png"
        />
      </div>
      <div
        className="mt-3 loading-text"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
